pyensign/ensign.py

```diff
@@ -423,17 +424,30 @@ async def destroy_topic(self, id: str) -> bool:
 
         Parameters
         ----------
-        id : str
-            The ID of the topic to destroy.
+        topic : str
+            The name or ID of the topic to destroy.
 
-        Returns
-        -------
-        bool
-            True if the request was accepted, False otherwise.
+        Raises
+        ------
+        EnsignTopicNotFoundError
+            If the topic was not found.
+
+        EnsignTopicDestroyError
+            If the topic could not be destroyed.
         """
 
+        # Resolve the topic ID from the string
+        try:
+            id = self.topics.resolve(topic)
+        except CacheMissError:
+            # If the ID is not in the cache, look it up from the server
+            id = await self.topic_id(topic)
+
         _, state = await self.client.destroy_topic(id)
-        return state == topic_pb2.TopicState.DELETING
+        if state != TopicState.DELETING:
+            raise EnsignTopicDestroyError(
+                "failed to destroy topic, topic state is {}".format(state)
+            )
```

Comment:
Changed this to raise an exception rather than return a bool which is a bit more pythonic I think.

Reply:
I like this change - it is more informative as well